cmd/osm-bootstrap/crds/config_mesh_root_certificate.yaml

```diff
@@ -106,16 +106,48 @@ spec:
                           type: string
                         token:
                           description: Token used by the mesh control plane
-                          type: string
+                          type: object
+                          required:
+                            - secretRef
+                          properties:
+                            secretRef:
+                              description: Reference to the kubernetes secret storing the vault token
+                              type: object
+                              required:
+                                - name
```

Comment:
you also need the key to know how to access the data.

 Tresor does not need the key

Reply:
Assuming that the tresor cert is self-created, although it may be a good user experience to not have that requirement. In reality it should probably not be an opaque secret but instead be a tls cert type.

For vault, ya we can do that, although if a user already has the token as a secret it might be nice for them to reuse it.

Reply:
The Tresor cert can be created by the user or OSM. Currently, the `osm-ca-bundle` is an opaque secret unless we plan to no longer create an `osm-ca-bundle`.

Reply:
Updated the Vault token secretRef to be a secretKeyRef